fastchat/data/optional_replace.py

```diff
@@ -0,0 +1,82 @@
+"""
+Do optional replace of bos/eos/pad/unk.
+
+Usage:
+python3 -m fastchat.data.optional_replace --in input.json --out output.json --model-name-or-path <your_token_path>
+
+Requirement:
+pip3 install transformers tqdm
+"""
+import argparse
+import json
+import traceback
+
+import transformers
+from tqdm import tqdm
+
+
+def replace_special_tokens(
+    tokenizer: transformers.PreTrainedTokenizer, text: str
+) -> str:
+    if not text:
+        return text
+
+    def _insert_vline(token: str) -> str:
+        if len(token) < 2:
+            return " "
+        elif len(token) == 2:
+            return f"{token[0]}|{token[1]}"
+        else:
+            return f"{token[:1]}|{token[1:-1]}|{token[-1:]}"
+
+    if tokenizer.bos_token:
+        text = text.replace(tokenizer.bos_token, _insert_vline(tokenizer.bos_token))
```

Comment:
@Trangle

1. This incurs a lot of redundant computation because this function is called for every sentence. Could you extract this common expression out of this function? You can define a global variable like
```tokenizer_bos_token_target = _insert_vline(tokenizer.bos_token))```

2. May I ask why your training data contain these special tokens?